syntax = "proto3";

package freshCache;

service DBService {
  rpc Put (DBPutRequest) returns (DBPutResponse);
  rpc Get (DBGetRequest) returns (DBGetResponse);
  rpc Delete (DBDeleteRequest) returns (DBDeleteResponse);
  rpc GetLoad(DBGetLoadRequest) returns (DBGetLoadResponse); 
  rpc GetReadCount(DBGetReadCountRequest) returns (DBGetReadCountResponse);
  rpc GetWriteCount(DBGetWriteCountRequest) returns (DBGetWriteCountResponse);
  rpc StartRecord(DBStartRecordRequest) returns (DBStartRecordResponse);
}

message DBPutRequest {
  string key = 1;
  bytes value = 2;
  float ew = 3;
}

message DBPutResponse {
  bool success = 1;
}

message DBGetRequest {
  string key = 1;
}

message DBGetResponse {
  bytes value = 1;
  bool found = 2;
}

message DBDeleteRequest {
  string key = 1;
}

message DBDeleteResponse {
  bool success = 1;
}

message DBGetLoadRequest {

}

message DBGetLoadResponse {
  int32 load = 1;
  bool success = 2;
}

message DBStartRecordRequest {

}

message DBStartRecordResponse {
  bool success = 1;
}

message DBGetReadCountRequest {}

message DBGetReadCountResponse {
  int64 read_count = 1;
  bool success = 2;
}

message DBGetWriteCountRequest {}

message DBGetWriteCountResponse {
  int64 write_count = 1;
  bool success = 2;
}